scripts/generate/generate.sh

```diff
@@ -0,0 +1,139 @@
+     You can provide it here or through json config(--config-path). If this is provided, the filter in json config will be ignored. (optional)
+    
+    -o  | --override            Strings used to override the output. It should follow the format:{directive:bitmask00|bitmask01...,bitmask10|bitmask11...}.
+     An example is --override log_format:ngxHTTPMainConf|ngxConf2More,ngxStreamMainConf|ngxConf2More.
+     To use | and , in command line, you may need to enclose your input in quotes, i.e. --override 'directive:mask1,mask2,...'.
+     You can provide it here or through json config(--config-path). If this is provided, the override in json config will be ignored. (optional)
+    
+    -mc | --match-func-comment  The code comment for generated matchFunc.
+	 You can add some explanations like which modules included in it. Normally it should start with match-func-name.
+	 If this is provided, the matchFuncComment in json config will be ignored. (optional)
+    
+    -p  | --path                Path to a directory in the repository containing the source code of the nginx module. (optional)
+    
+    --url                       Url used for git clone. (required)
+EOF
+}
+
+while [ ! $# -eq 0 ]; do
+    case "$1" in
+        --help | -h)
+            help
+            exit 0
+            ;;
+        --branch | -b)
+            branch="$2"
+            shift
+            ;;
+        --config-path | -c)
+            genArgs+=("--config-path=$2")
+            shift
+            ;;
+        --directive-map-name | -d)
+            genArgs+=("--directive-map-name=$2")
+            shift
+            ;;
+        --match-func-name | -mn)
+            genArgs+=("--match-func-name=$2")
+            shift
+            ;;
+        --filter | -f)
+            genArgs+=("-filter=$2")
+            shift
+            ;;
+        -override | -o)
+            genArgs+=("-override=\"$2\"")
+            shift
+            ;;
+        --match-func-comment | --mc)
+            genArgs+=("--match-func-comment=\"$2\"")
+            shift
+            ;;
+        --path | -p)
+            sub_path="$2"
+            shift
+            ;;
+        --url)
+            url="$2"
+            shift
+            ;;
+        *)
+            printf "Invalid option: %s\n" "$1" 1>&2
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [ "$url" = "" ]; then
+    echo "url can't be empty"
+    exit 1
+fi
+
+
+tmp=$(mktemp -d)
+cleanup() {
+    rm -rf "$tmp"
+}
+trap cleanup EXIT
+
+genArgs+=("--src-path=$tmp")
+
+if [ "$sub_path" = "" ]; then
+    git clone "$url" "$tmp" --depth 1 --branch "$branch" -q 1>&2
+else
+    (
+        git clone "$url" "$tmp" --no-checkout --depth 1 --branch "$branch" -q 1>&2 &&
+        # sparse-checkout doesn't support -q option
+        git -C "$tmp" sparse-checkout init --cone 1>&2  &&
+        git -C "$tmp" sparse-checkout set "$sub_path" 1>&2 &&
+        git -C "$tmp" checkout -q 1>&2
+    )
```

Comment:
Nit: You can do `cd "$tmp"` between the `()` as the directory will only be changed between the parentheses as those create a subshell.

Reply:
Awesome! Applied: ` cd "$tmp" ` after git clone